File: src/test/testdata/inspections/unsafeIterators/ObjectMap.java
package main;

import com.badlogic.gdx.utils.ObjectMap;

import java.util.Iterator;

class ObjectMapTest {

  void test() {
    ObjectMap<ObjectMap<String, ObjectMap<Object, Object>>, Object> map1 = new ObjectMap<ObjectMap<String, ObjectMap<Object, Object>>, Object>();
    ObjectMap<String, Object> map2 = new ObjectMap<String, Object>();

    for (ObjectMap.Entry<ObjectMap<String, ObjectMap<Object, Object>>, Object> m: <warning>map1</warning>) {
      for (ObjectMap.Entry<String, ObjectMap<Object, Object>> e: <warning>m.key</warning>) {
        for (Object p: <warning>e.value</warning>) {

        }
      }
    }

    Iterator i = <warning>map2.keys()</warning>;
    Iterator i2 = <warning>map2.values()</warning>;
    Iterator i3 = <warning>map2.entries()</warning>;
    Iterator i4 = <warning>map2.iterator()</warning>;
  }

}

File: src/test/testdata/inspections/unsafeIterators/ObjectMap.kt
package main

import com.badlogic.gdx.utils.ObjectMap

@Suppress("UNUSED_VARIABLE", "UNCHECKED_CAST", "USELESS_CAST")
class ObjectMapTestKotlin {

  fun test() {
    val map1 = ObjectMap<ObjectMap<String, ObjectMap<Any, Any>>, Any>()
    val map2 = ObjectMap<String, Any>()

    for (m in <warning>map1</warning>) {
      for (e in <warning>m.key</warning>) {
      for (p in <warning>e.value</warning>) {

    }
    }
    }

    val i = <warning>map2.keys()</warning>
    val i2 = <warning>map2.values()</warning>
    val i3 = <warning>map2.entries()</warning>
    val i4 = <warning>map2.iterator()</warning>
  }

}

